File: 1.Easy/#392Is_Subsequence/solution.c
bool isSubsequence(char * s, char * t){
    int i = 0, j = 0;
    if (strlen(s) == 0) {
        return true;
    }
    if (strlen(t) == 0) {
        return false;
    }
    
    while (j < strlen(t)) {
        if (s[i] == t[j]){
            i += 1;
            if (i == strlen(s)){
                return true;
            }
        }
        j += 1;
    }
    return false;
    
}

File: 1.Easy/#392Is_Subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l = [-1]
        for c in s:
            j = t.find(c, l[-1]+1)
            if j == -1:
                return False
            l += [j]
            
        return True